modules/apps/journal/journal-content-web/src/main/java/com/liferay/journal/content/web/internal/portlet/toolbar/contributor/JournalContentPortletToolbarContributor.java

```diff
@@ -137,10 +138,12 @@ private void _addPortletTitleAddJournalArticleMenuItems(
 		if (journalContentPortletInstanceConfiguration.
 				sortStructuresByByName()) {
 
-			ddmStructures = _ddmStructureService.getStructures(
-				themeDisplay.getCompanyId(), currentAndAncestorSiteGroupIds,
-				_portal.getClassNameId(JournalArticle.class), QueryUtil.ALL_POS,
-				QueryUtil.ALL_POS, new StructureCreateDateComparator());
+			ddmStructures = ListUtil.copy(
+				_ddmStructureService.getStructures(
+					themeDisplay.getCompanyId(), currentAndAncestorSiteGroupIds,
+					_portal.getClassNameId(JournalArticle.class),
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS,
+					new StructureCreateDateComparator()));
```

Comment:
If the problem was that the collection was unmodifiable, why not use `ListUtil.sort`? That method makes a copy if necessary.

Also, that would allow you to inline much of the code:
```
ddmStructures = ListUtil.sort(
        _ddlStructureService.getStructures(...),
        (ddmStructure1, ddmStructure2) -> {
                 ...
        });
```

Reply:
I didn't know that one, nice!